in_toto/models/layout.py

```diff
@@ -126,12 +173,376 @@ def read(data):
+      all steps with the passed name.
+
+    <Arguments>
+      step_name:
+              A step name.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError
+              If the passed step name is not a string.
+
+    """
+    securesystemslib.schema.AnyString().check_match(step_name)
+
+    for step in self.steps:
+      if step.name == step_name:
+        self.steps.remove(step)
+
+
+  def get_inspection_name_list(self):
+    """
+    <Purpose>
+      Return list of inspection names in the order in which they are listed in
+      the layout.
+
+    <Returns>
+      A list of the inspection names.
+
+    """
+    inspection_names = []
+    for inspection in self.inspect:
+      inspection_names.append(inspection.name)
+
+    return inspection_names
+
+
+  def get_inspection_by_name(self, inspection_name):
+    """
+    <Purpose>
+      Return the first inspection in the layout's list of inspections
+      identified by the passed inspection name.
+
+      NOTE: Inspection names must be unique within a layout, which is enforced
+      by they layout's validate method. However, if validate has not been
+      called, there may be multiple inspections with the same name. In that
+      case only the first inspection with the passed name is returned.
+
+    <Arguments>
+      inspection_name:
+              An inspection name.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError
+              If the passed inspection name is not a string.
+
+    <Returns>
+      An inspection object.
+
+    """
+    securesystemslib.schema.AnyString().check_match(inspection_name)
+
+    for inspection in self.inspect: # pragma: no branch
+      if inspection.name == inspection_name:
+        return inspection
+
+
+  def remove_inspection_by_name(self, inspection_name):
+    """
+    <Purpose>
+      Remove all inspections from the layout's list of inspections identified
+      by the passed inspection name.
+
+      NOTE: Inspection names must be unique within a layout, which is enforced
+      by they layout's validate method. However, if validate has not been
+      called, there may be multiple inspections with the same name. The
+      method removes all inspections with the passed name.
+
+    <Arguments>
+      inspection_name:
+              An inspection name.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError
+              If the passed inspection name is not a string.
+
+    """
+    securesystemslib.schema.AnyString().check_match(inspection_name)
+
+    for inspection in self.inspect:
+      if inspection.name == inspection_name:
+        self.inspect.remove(inspection)
+
+
+  def get_functionary_key_id_list(self):
+    """
+    <Purpose>
+      Return a list of the functionary keyids from the layout's keys
+      dictionary.
+
+    <Returns>
+      A list of functionary keyids.
+
+    """
+    keyid_list = []
+    for keyid in list(self.keys.keys()):
+      keyid_list.append(keyid)
```

Comment:
I know this goes against the our guidelines, but a list comprehension here would be way more readable imho.

Reply:
I guess that's a matter of taste. And while I'm a big fan of list comprehensions, I think we should stick to our guidelines.

Reply:
Why is `keyid_list` needed?  Doesn't the following work?
```Python
return list(self.keys.keys())
```

Note: the list() is needed for Python 2+3 compatibility, otherwise `return self.keys.keys()` should be fine.

Reply:
Good thinking Vlad! :)